data_steward/cdr_cleaner/cleaning_rules/sandbox_and_remove_pids.py

```diff
@@ -1,120 +1,148 @@
+        """
+        if self.person_table_list:
+            return self.person_table_list
+
+        person_table_query = PERSON_TABLE_QUERY.render(project=self.project_id,
+                                                       dataset=self.dataset_id)
+        person_tables = client.query(person_table_query).result()
+        # utils.bq.query(person_table_query).get(
+        #     TABLE_NAME_COLUMN).to_list()
+        # exclude mapping tables from list, to be removed after all cleaning rules
+        self.person_table_list = [
+            table.get('table_name')
+            for table in person_tables
+            if '_mapping' not in table.get('table_name')
+        ]
+
+        return self.person_table_list
+
+    def get_sandbox_queries(self, lookup_table: str = None) -> list:
+        """
+        Returns a list of queries of all tables to be added to the datasets sandbox. These tables include all rows from all
+        effected tables that include PIDs that will be removed by a specific cleaning rule.
+
+        :param lookup_table: name of the lookup table
+        :return: list of CREATE OR REPLACE queries to create tables in sandbox
+        """
+        if self.sandbox_dataset_id is None:
+            raise RuntimeError(
+                f"sandbox_dataset_id is None.  This is not allowed.")
+
+        # person_tables_list = self.get_tables_with_person_id()
+        queries_list = []
+
+        for table in self.person_table_list:
+            queries_list.append({
+                cdr_consts.QUERY:
+                    SANDBOX_QUERY.render(
+                        dataset=self.dataset_id,
+                        project=self.project_id,
+                        table=table,
+                        sandbox_dataset=self.sandbox_dataset_id,
+                        intermediary_table=self.sandbox_table_for(table),
+                        lookup_table=lookup_table)
+            })
+
+        return queries_list
+
+    def get_remove_pids_queries(self, lookup_table=None):
+        """
+        Returns a list of queries in which the table will be truncated with clean data, ie: all removed PIDs from all
+        datasets based on a cleaning rule.
+
+        :param lookup_table: name of the lookup table
+        :return: list of select statements that will truncate the existing tables with clean data
+        """
+        # person_tables_list = self.get_tables_with_person_id()
+        queries_list = []
+
+        for table in self.person_table_list:
+            queries_list.append({
+                cdr_consts.QUERY:
+                    CLEAN_QUERY.render(project=self.project_id,
+                                       dataset=self.dataset_id,
+                                       table=table,
+                                       sandbox_dataset=self.sandbox_dataset_id,
+                                       lookup_table=lookup_table)
+            })
+
+        return queries_list
```

Comment:
Should this define get_query_specs and other methods?

Reply:
I purposely left this as an abstract class so it cannot be instantiated by iteself.  It must be extended to be useful.  The extending rule should define `get_query_specs` and any other methods that are required to make this class "concrete".

What are your thoughts?

Reply:
Yes, that makes sense. As long as this rule isn't called by itself, it can stay this way.